plugins-builtin/extractcode-libarchive-system_provided/src/extractcode_libarchive/__init__.py

```diff
@@ -28,36 +28,61 @@
 from os.path import abspath
 from os.path import dirname
 from os.path import join
+from os import path
 
 from plugincode.location_provider import LocationProviderPlugin
 from plugincode.location_provider import location_provider_impl
 
 import platform
 
 class LibarchivePaths(LocationProviderPlugin):
-	def get_locations(self):
-		curr_dir = dirname(abspath(__file__))
-		distribution=platform.linux_distribution()[0]	
-	
-		# List of various major distributions consisting of flavors
-		debian_based_distro=['Ubuntu','Mint','debian']
+    def get_locations(self):
+
+        '''This defines the various mainstream system available 
+        freebsd(unix derivative),Linux(unix alternative)'''
+        
+        mainstream_system = platform.system()
+
+        if mainstream_system == 'Linux' :
+            
+            distribution=platform.linux_distribution()[0]	
 	
-		rpm_based_distro=['Fedora','redhat']
+            '''List of linux distributions consisting of flavors.'''
+		
+            debian_based_distro=['Ubuntu','Mint','debian']
+	    rpm_based_distro=['Fedora','redhat']
 
-		if distribution in debian_based_distro:
+            if distribution in debian_based_distro:
 		
-			lib_dir = '/usr/lib/x86_64-linux-gnu'
+                lib_dir = '/usr/lib/x86_64-linux-gnu'
 	
-		elif distribution in rpm_based_distro:
-		
-			lib_dir = '/usr/lib64'
+            elif distribution in rpm_based_distro:
+
+                lib_dir = '/usr/lib64'
 
-		else:
-			#User defined if installation directory differs
-			lib_dir = '/usr'
+            else:
 
-		locations = {
-			'extractcode.libarchive.libdir': lib_dir,
-			'extractcode.libarchive.dll': join(lib_dir, 'libarchive.so.13'),
-		}
-		return locations
+                '''User defined flavor if installation directory differs'''
```

Comment:
Why do you use triple quotes there?  a plain # would be better? also what do you mean exactly with this comment?

Reply:
If the user has defined some other path of installation or some different flavor from unusual.